pkg/pillar/cmd/zedkube/lease.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build kubevirt
+
+package zedkube
+
+import (
+	"context"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/kubeapi"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+)
+
+func handleLeaderElection(ctx *zedkubeContext) {
```

Comment:
Personally I would prefer we stop using the `<microservice-name>Context` pattern for new microservices.
Instead, zedkube can be a struct with methods:

```
type zedkube struct {
 ....
}

func (z *zedkube) handleLeaderElection {
 ...
}
```

We have it like that in zedrouter, nim and few other microservices. IMHO it reads better.

Reply:
i'll see if i can convert

Reply:
thanks for making this change